### PR TITLE
Add some tuning options for router and controller

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Comment by grs** [comment]
Fixes issues #464 and #465

File: api/types/client.go [hunk 1]
```diff
@@ -35,26 +35,41 @@ type RouterLogConfig struct {
...
+	User                string
+	Password            string
+	Ingress             string
+	ConsoleIngress      string
+	Replicas            int32
+	SiteControlled      bool
+	Annotations         map[string]string
+	Labels              map[string]string
+	Router              RouterOptions
+	Controller          Tuning
```
Comment (ajssmith) [inline api/types/client.go:72]: Minor or we could do in the future. For symmetry, suggest introducing ControllerOptions that embeds Tuning struct as down the road we may have logging, debugging settings for the Controller as well.

**Review by ajssmith — APPROVED** [review]
lgtm

Built locallly, manually tested/verified setting resource/limit settings and affinity values. Ran cluster tests. Did not verify that affinity setting resulted in placement result.

File: api/types/client.go (hunk 1 above)
Comment (grs) [reply api/types/client.go]: Done

File: client/router_create.go [hunk 2]
```diff
@@ -226,6 +236,36 @@ func (cli *VanClient) GetVanControllerSpec(options types.SiteConfigSpec, van *ty
...
+	}
+	if options.NodeSelector != "" {
+		spec.NodeSelector = utils.LabelToMap(options.NodeSelector)
+	}
+	if options.Cpu != "" {
+		cpuQuantity, err := resource.ParseQuantity(options.Cpu)
+		if err == nil {
+			spec.CpuRequest = &cpuQuantity
+		} else {
+			err = fmt.Errorf("Invalid value for cpu: %s", err)
```
[the remark this answers is not in the document]
Comment (grs) [reply client/router_create.go:256]: Yes, you are right. I'll have another go!

[the remark this answers is not in the document]
Comment (fgiorgetti) [reply client/router_create.go]: Looks great.

File: client/site_config_inspect.go [hunk 3]
```diff
@@ -118,45 +118,91 @@ func (cli *VanClient) SiteConfigInspectInNamespace(ctx context.Context, input *c
...
+	if controllerMemory, ok := siteConfig.Data["controller-memory"]; ok && controllerMemory != "" {
+		result.Spec.Controller.Memory = controllerMemory
+	}
+	if controllerNodeSelector, ok := siteConfig.Data["controller-node-selector"]; ok && controllerNodeSelector != "" {
+		result.Spec.Controller.NodeSelector = controllerNodeSelector
+	}
+	if controllerAffinity, ok := siteConfig.Data["controller-affinity"]; ok && controllerAffinity != "" {
+		result.Spec.Controller.Affinity = controllerAffinity
+	}
+	if controllerAntiAffinity, ok := siteConfig.Data["controller-anti-affinity"]; ok && controllerAntiAffinity != "" {
```
[the remark this answers is not in the document]
Comment (grs) [reply client/site_config_inspect.go:177]: Agreed, I'm working on a separate PR for that now.

File: api/types/types.go [hunk 4]
```diff
@@ -140,10 +141,19 @@ const (
 	TokenCost                   string = BaseQualifier + "/cost"
 	UpdatedAnnotation           string = InternalQualifier + "/updated"
 	AnnotationExcludes          string = BaseQualifier + "/exclude-annotations"
+	LabelExcludes               string = BaseQualifier + "/exclude-labels"
 	ComponentAnnotation         string = BaseQualifier + "/component"
+	SiteControllerIgnore        string = InternalQualifier + "/site-controller-ignore"
```
[the remark this answers is not in the document]
Comment (grs) [reply api/types/types.go:146]: Agreed, will fix.

File: client/site_config_inspect.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (grs) [reply client/site_config_inspect.go:177]: See https://github.com/skupperproject/skupper/pull/484 for this

**Pull request merged**